### PR TITLE
String lab

### PR DESCRIPTION
**Pull request by Misseswaddles**
Completed string formatting lab

### CODE REVIEW

**Comment by PythonCHB** [comment]
`finalString = '{}{}{}{}'.format(stringPieces[0],`

so this is doing for "instertions" -- butyou can use format specifiers in those l{}, rather than pre-formatting the contents.

and the idea behind supporting multiple length tuple was that one piece of code could support ANY length tuple -- not that you'd write special code for each length.

**Comment by PythonCHB** [comment]
there are conflicts in these files -- I suspect it's because they changed in your master, plus getting merged into the main repo master.

try, on your laptop:

git checkout stringLab
git merge master

resolve any conflicts if there are any.

then

git push

**Comment by Misseswaddles** [comment]
Yikes,

What a hot mess! I am thinking to withdraw the pull request and re-do the stringlab on a clean separate branch?

**Comment by PythonCHB** [comment]
Sure — sometimes a redo is the easiest way to go :-)

File: students/C_Farris/session03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+
+def addUsingInsert(myFruit):
+    myFruit.insert(0, 'Pineapple')
+    print(myFruit)
+
+
+def addToBeginningOfList(myFruit):
+    singleList = ['Mango']
+    myFruit = singleList + myFruit
```
Comment (PythonCHB) [inline students/C_Farris/session03/list_lab.py]: did this work? it shouldn't :-(

```
In [1]: singleList = ['Mango']                                                  

In [2]: singleList + "Banana"                                                   
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-2234c9855b46> in <module>
----> 1 singleList + "Banana"

TypeError: can only concatenate list (not "str") to list
```

File: students/C_Farris/session03/list_lab.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+def displayPfruits(myFruit):
+    for x in myFruit:
+        if x.lower().startswith('p'):
+            print(x)
+        else:
+            print(x + "  doesn't start with p")
+
+
+def addUsingInsert(myFruit):
+    myFruit.insert(0, 'Pineapple')
```
Comment (PythonCHB) [inline students/C_Farris/session03/list_lab.py]: yup!

File: students/C_Farris/session03/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    newFruit = newFruit.strip().capitalize()  # strip any whitespace
+    myFruit.append(newFruit)
+    print(myFruit)
+
+
+def askForANumber():
+    fruitNumb = 100
+    fruitLength = len(myFruit) - 1
+    textInput = "Please select an integer from 0-{} ==>".format(fruitLength)
+    while not fruitNumb <= fruitLength:
```
Comment (PythonCHB) [inline students/C_Farris/session03/list_lab.py]: nice!

File: students/C_Farris/session03/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    for x in myFruit:
+        question = ""
+        verdict = ''
+        while verdict != 'yes' and verdict != 'no':
+            question = "Do you like {}? Yes or No? ==>  ".format(x.lower())
+            verdict = input(question)
+            verdict = verdict.strip().lower()
+            if verdict == 'no':
+                fruitToDelete.append(x)
+    findAndDeleteFruit(fruitToDelete)
```
Comment (PythonCHB) [inline students/C_Farris/session03/list_lab.py]: nice! -- though the other option is to build up a new list of the stuff NOT to delete as you go :-)

**Comment by PythonCHB** [comment]
you've got conflicts :-(

try:

git pull upstream master

and see if you can fix them on your end first.

**Comment by Misseswaddles** [comment]
Hi Chris,

I think this is carryover from my original mess up with branching. I
submitted the list_lab on the List_lab branch but it had conflicts and that
pull request was closed. The StringLab branch is where I messed everything
up, I think, as it had an incompleted version of list_lab.py on it. So, I
will just need to stop by office hours to finish cleaning up the mess.I
have SO learned my lesson about being careful what to touch on branches!!!

On the subject of Stringlab.py, I started from scratch on the string lab
exercise and created a new pull request on the Master branch. It doesn't
appear to have any conflicts so we are good. The list_lab.py is still
lingering in the list_lab branch but I need to clean up the StringLab
branch before I will do a new pull request!

Sorry about this!

Best,
Carol

**Comment by PythonCHB** [comment]
Did this all get turned in in another PR? that is, can we close this?

-CHB

**Pull request closed without merging**